### ISSUE TITLE
Training process not starting but terminates with saving weights

### ISSUE THREAD

**Issue by hasiffirdos** [AlexeyAB/darknet#4025]
hope you are doing well. previously same code was working but now it stopped working. 
**I run this Command:**

`./darknet detector train data/obj.data ./cfg/yolov3-tiny-obj.cfg /content/darknet/yolov3-tiny.weights -dont_show`

**and output is:** 
yolov3-tiny-obj
   layer   filters  size/strd(dil)      input                output
   0 conv     16       3 x 3/ 1    480 x 480 x   3 ->  480 x 480 x  16 0.199 BF
   1 max                2x 2/ 2    480 x 480 x  16 ->  240 x 240 x  16 0.004 BF
   2 conv     32       3 x 3/ 1    240 x 240 x  16 ->  240 x 240 x  32 0.531 BF
   3 max                2x 2/ 2    240 x 240 x  32 ->  120 x 120 x  32 0.002 BF
   4 conv     64       3 x 3/ 1    120 x 120 x  32 ->  120 x 120 x  64 0.531 BF
   5 max                2x 2/ 2    120 x 120 x  64 ->   60 x  60 x  64 0.001 BF
   6 conv    128       3 x 3/ 1     60 x  60 x  64 ->   60 x  60 x 128 0.531 BF
   7 max                2x 2/ 2     60 x  60 x 128 ->   30 x  30 x 128 0.000 BF
   8 conv    256       3 x 3/ 1     30 x  30 x 128 ->   30 x  30 x 256 0.531 BF
   9 max                2x 2/ 2     30 x  30 x 256 ->   15 x  15 x 256 0.000 BF
  10 conv    512       3 x 3/ 1     15 x  15 x 256 ->   15 x  15 x 512 0.531 BF
  11 max                2x 2/ 1     15 x  15 x 512 ->   15 x  15 x 512 0.000 BF
  12 conv   1024       3 x 3/ 1     15 x  15 x 512 ->   15 x  15 x1024 2.123 BF
  13 conv    256       1 x 1/ 1     15 x  15 x1024 ->   15 x  15 x 256 0.118 BF
  14 conv    512       3 x 3/ 1     15 x  15 x 256 ->   15 x  15 x 512 0.531 BF
  15 conv     21       1 x 1/ 1     15 x  15 x 512 ->   15 x  15 x  21 0.005 BF
  16 yolo
[yolo] params: iou loss: mse, iou_norm: 0.75, cls_norm: 1.00, scale_x_y: 1.00
  17 route  13
  18 conv    128       1 x 1/ 1     15 x  15 x 256 ->   15 x  15 x 128 0.015 BF
  19 upsample                 2x    15 x  15 x 128 ->   30 x  30 x 128
  20 route  19 8
  21 conv    256       3 x 3/ 1     30 x  30 x 384 ->   30 x  30 x 256 1.593 BF
  22 conv     21       1 x 1/ 1     30 x  30 x 256 ->   30 x  30 x  21 0.010 BF
  23 yolo
[yolo] params: iou loss: mse, iou_norm: 0.75, cls_norm: 1.00, scale_x_y: 1.00
Total BFLOPS 7.255 
 Allocate additional workspace_size = 33.18 MB 
Loading weights from /content/darknet/yolov3-tiny.weights...
 seen 64 
Done! Loaded 24 layers from weights-file 
Learning Rate: 0.001, Momentum: 0.9, Decay: 0.0005
Saving weights to backup//yolov3-tiny-obj_final.weights

thanks

**Comment by AlexeyAB** [comment]
Use correct max_batches= in cfg-file.

**Comment by hasiffirdos** [comment]
Thank you so much @AlexeyAB for your time. 
yes, i am using this cfgs for network.


[net]
batch=64
subdivisions=2
width=416
height=416
channels=3
momentum=0.9
decay=0.0005
angle=0
saturation = 1.5
exposure = 1.5
hue=.1

learning_rate=0.001
burn_in=1000
`max_batches = 5000`
policy=steps
`steps=4000,4500`
scales=.1,.1

[convolutional]
batch_normalize=1
filters=16
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=32
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=64
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=128
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=256
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=512
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=1

[convolutional]
batch_normalize=1
filters=1024
size=3
stride=1
pad=1
activation=leaky

###########

[convolutional]
batch_normalize=1
filters=256
size=1
stride=1
pad=1
activation=leaky

[convolutional]
batch_normalize=1
filters=512
size=3
stride=1
pad=1
activation=leaky

[convolutional]
size=1
stride=1
pad=1
filters=21
activation=linear



[yolo]
mask = 3,4,5
anchors= 4.7439693,3.56052632,  5.96004796,4.44855441,  6.93922585,5.17026503,  7.84043062,5.85085526,  8.85860809,6.59607474,  10.25595052,7.63565918


classes=2
num=6
jitter=.3
ignore_thresh = .7
truth_thresh = 1
random=1

[route]
layers = -4

[convolutional]
batch_normalize=1
filters=128
size=1
stride=1
pad=1
activation=leaky

[upsample]
stride=2

[route]
layers = -1, 8

[convolutional]
batch_normalize=1
filters=256
size=3
stride=1
pad=1
activation=leaky

[convolutional]
size=1
stride=1
pad=1
filters=21
activation=linear

[yolo]
mask = 0,1,2
anchors= 4.7439693,3.56052632,  5.96004796,4.44855441,  6.93922585,5.17026503,  7.84043062,5.85085526,  8.85860809,6.59607474,  10.25595052,7.63565918

classes=2
num=6
jitter=.3
ignore_thresh = .7
truth_thresh = 1
random=1
